include/signal.h

```diff
@@ -542,7 +553,12 @@ int  sigwaitinfo(FAR const sigset_t *set, FAR struct siginfo *value);
  ********************************************************************************/
 
 #ifndef __SIGSET_T_DEFINED
-typedef uint32_t sigset_t;
+struct sigset_s
+{
+  uint32_t _elem[_SIGSET_NELEM];
+};
+
```

Comment:
@patacongo BTW, are we sure that `#else /* __INCLUDE_SIGNAL_H */` is needed at all?
What is the use case for it? I'm asking because `_SIGSET_NELEM` is defined in `#ifndef __INCLUDE_SIGNAL_H` block and if it is visible here that automatically means that `sigset_t` is already available

Reply:
I did that years ago because there were circular dependencies that could not be fixed in the normal way.  I can't remember what that was now but ASFAIK it would break the build to remove that code.

But perhaps things have changed.

This is beyond the scope of the current change.  We should address it in any situation and hold up this unrelated change.   Let's not let the scope of this PR creep beyond increasing the number of signals.  You can test if it is necessary by just removing the #else and see what breaks.  Perhaps there is a better solution.

Reply:
Yes and No. `_SIGSET_NELEM` should not be visible here as it is defined in `#ifndef __INCLUDE_SIGNAL_H` scope above. I mean that it is not `uint32_t` anymore.

Reply:
So let's add
```
efine MAX_SIGNO 63 /* Highest valid signal number *
#define _NSIG (MAX_SIGNO + 1) /* Biggest signal number + 1 */
#define _SIGSET_NELEM ((_NSIG + 31) >> 5)
```
here as well

Reply:
We only get to that logic IF __INCLUDE_SIGNAL_H  is defined.  If so, yes _SEGSET_NELEM should also be defined (it builds!), right?  But then so should __SIGSET_T_DEFINED.

I don't remember why this is there at all.  It would require some separate investigation.

Reply:
Well.. it builds fine with the #else condition removed!  I will remove it now.

Reply:
Great! I was expecting it